sample/src/main/java/co/nimblehq/sample/ui/surveylist/SurveyAdapter.kt

```diff
@@ -31,8 +33,15 @@ class SurveyAdapter : RecyclerView.Adapter<SurveyAdapter.SurveyViewHolder>() {
     }
 
     class SurveyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        fun bind(value: String) {
-            (itemView as TextView).text = value
+        fun bind(value: SurveyModel) {
+            (itemView as TextView).text = value.surveyTitle
+            itemView.setOnClickListener {
+                val intent = Intent(itemView.context, SurveyDetailActivity::class.java).apply {
+                    putExtra("id", value.surveyId)
+                    putExtra("title", value.surveyTitle)
+                }
+                itemView.context.startActivity(intent)
+            }
```

Comment:
I understand that this is the sample app but with our current projects, we should create the callback and return `SurveyModel` to handle outside of Adapter's scope. The adapter should be purposed for ViewHolder and bind data to ViewHolder.

Reply:
@sleepylee What do you think about this? Do we need to follow up all our rule with this `sample-app` ?

Reply:
agree, please handle this step in activity instead

Reply:
@doannimble I believe falling back and handling adapter events in the activity or fragment from the adapter is a common best practice we should follow, even for a simple sample app.

Reply:
Actually, MVVM architecture is what to be optional in sample app

Reply:
not really necessary super clean code in sample-app. It's a sample and the stakeholders are waiting more for what they sdk can do, we don't really care much about the clean code on the sample app.
Please prioritize correctly, @hoangnguyen92dn @luongvo @doannimble

Reply:
the sample sometimes needs to be clear and straight forward actually; I would have thought that we did not have to implement ViewModel also, because more abstraction == hard for the new dev to understand what do they have to do to use the sdk, right?

Reply:
@sleepylee yes, MVVM is optional imo, but falling back the event to activity or fragment from adapter is a basic concept we should follow I believe

Reply:
for example, now if a new devs read our docs for the survey fetching; they will have to go through the whole flow, down to the Repository to find the implementation code...imo; our sample-app discouraged them

Reply:
for @luongvo question: if it's about the correctness: does this already work correctly? Is sending callback making it more abstracted or harder to understand? 🤔

Reply:
@luongvo I think it a best practice, not a basic concept.

Reply:
ok, if you guys think it's not so important to follow in the sample app, I would agree to ignore this issue.